### ISSUE TITLE
Errors running on Linux (some fatal)

### ISSUE THREAD

**Issue by trvrfrd** [thenewinquiry/bailbloc#19]
When trying to start the app from the command line on Linux, it blows up on `desktop/main.js:219` because no `activeTrayImage` is assigned unless the platform is `darwin` or `win32`:

```
Uncaught Exception:
TypeError: Error processing argument at index 0, conversion failure from undefined
    at App.app.on (/home/trevor/oss/bailbloc/desktop/main.js:219:10)
    at emitTwo (events.js:111:20)
    at App.emit (events.js:194:7)
```

This assignment happens in an if statement on line 43 of the same file. Changing either branch of the if statement to evaluate to true (so, using either the Mac or Windows tray images) seems to get things up and running, although an error is still thrown when checking for updates: 
z
```
[12:23:35.518] [error] Cannot check for updates: Error: Cannot find latest-linux.yml in the latest release artifacts (https://github.com/thenewinquiry/bailbloc/releases/download/v1.1.2/latest-linux.yml): HttpError: 404 Not Found
```

I'm happy to help try and fix this, but I don't really know what I'm doing when it comes to Electron 👀

**Comment by niftynei** [comment]
I had a few problems getting the app to build, [this](https://github.com/SimulatedGREG/electron-vue/issues/130#issuecomment-331993257) helped:

```
sudo apt install icnsutils graphicsmagick
npm i 7zip-bin-linux   #not sure if this is necessary ?
```

also, I found that I needed to run `electron-builder` from the node-modules bin, basically:

```
 ./node_modules/.bin/electron-builder --linux
# then, start the app with
./dist/linux-unpacked/bailbloc
```

I'm running into an error when starting the app tho 

```

TypeError: Error processing argument at index 0, conversion failure from undefined
    at App.app.on (/home/lisa/dev/bailbloc/desktop/dist/linux-unpacked/resources/app/main.js:219:10)
    at emitTwo (events.js:111:20)
    at App.emit (events.js:194:7)
```

**Comment by niftynei** [comment]
also submitted a patch #21 to add the 7zip-bin-linux dependency to package.json; idk if this is strictly necessary but it runs now so.

**Comment by Shaderboy** [comment]
Running into the same TypeError issue as above. Is that what your patch fixes, @niftynei?

**Comment by niftynei** [comment]
@Shaderboy yep!

**Comment by raspatan** [comment]
I am too noob to install from source. Will there be some sort of package built from this?

**Comment by ferristseng** [comment]
If you don't really care about the GUI, I made a gist so you can just call into the packaged miner:

https://gist.github.com/ferristseng/1d76ee8518724cd8f83772380c5c36cb

**Comment by bhtucker** [comment]
gist worked for me! also @niftynei :wave:

**Comment by trvrfrd** [comment]
Looks like this was addressed (more or less) in #20 which is merged now, so I'm going to go ahead and close this issue.
